### PR TITLE
Updated js-guidelines

### PR DESCRIPTION
**Pull request by RichardLitt**
My updates.

### CODE REVIEW

File: js-project-guidelines.md [hunk 1]
```diff
@@ -1,113 +1,147 @@
...
+  - [Why are you bothering with ES2015 and all this build setup?](#why-are-you-bothering-with-es2015-and-all-this-build-setup)
+  - [Do I have to use ES2015 and Babel?](#do-i-have-to-use-es2015-and-babel)
+  - [Do I have to bundle everything with webpack?](#do-i-have-to-bundle-everything-with-webpack)
+  - [Why are you doing this?](#why-are-you-doing-this)
+- [Contributing](#contributing)
+- [Code of Conduct](#code-of-conduct)
+- [References - Resources and good reads](#references---resources-and-good-reads)
+- [Acknowledgment](#acknowledgment)
+
+## Goals and Expectations
```
Comment (noffle) [inline js-project-guidelines.md]: Nit: just "Goals"

File: js-project-guidelines.md [hunk 2]
```diff
@@ -1,113 +1,147 @@
...
+- **Encourage Contribution**.
+- **Great UX** for everyone involved.
 
-- project stucture.
-- code style.
-- continuous integration.
-- tests.
-- tasks (asset pipeline, transpiling, releasing, etc).
-- dependency management.
+## The Guidelines
```
Comment (noffle) [inline js-project-guidelines.md]: Nit: just "Guidelines"

File: js-project-guidelines.md [hunk 3]
```diff
@@ -1,113 +1,147 @@
...
+However, we've added an extra linting rule: Enforce the use of [strict mode](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Strict_mode). This avoids issues we had when using ES2015 features outside of strict mode. We enforce this rule by using [eslint](http://eslint.org/) and extending [standard module](https://github.com/feross/standard) with the [eslint-config-standard](https://github.com/feross/eslint-config-standard).
 
-#### b) Test
+#### Test
 
-Since `js-ipfs` is meant to be both a Node.js and Browser app, we strongly recommend to have tests that run in both platforms, always. For most cases, we use [mocha](http://mochajs.org) to run write the tests and [karma](http://karma-runner.github.io) to automate the test execution in the browser. This solution has been extremely convinient.
+Since `js-ipfs` is meant to be both a Node.js and Browser app, we strongly recommend having tests that run in both platforms, always. For most cases, we use [mocha](http://mochajs.org) to run write the tests and [karma](http://karma-runner.github.io) to automate the test execution in the browser. This solution has been extremely convenient.
 
-#### c) Build
+#### Build
```
Comment (noffle) [inline js-project-guidelines.md]: Grammar nit: if we're using "Testing" in the heading above, all of these should be in the progressive tense too, right?

File: js-project-guidelines.md [hunk 4]
```diff
@@ -1,113 +1,147 @@
...
 6. Create a git tag
-7. Push to github
+7. Push to GitHub
 8. Publish to npm
 
-### Day to day operations...
+## aegir.js
 
-We've created a tool to help us achieve all of the above, feel free to also use it for your projects. Feedback is appreciated!
+We've created a module to help us achieve all of the above with minimal effort. Feel free to also use it for your projects. Feedback is appreciated!
```
Comment (noffle) [inline js-project-guidelines.md]: Nit: linkify https://github.com/dignifiedquire/aegir

File: js-project-guidelines.md [hunk 5]
```diff
@@ -1,113 +1,147 @@
...
+    "release": "aegir-release"
   }
 }
 ```
 
 You also need to add it your `devDependencies` by running:
 
 ```sh
-$ npm install --save-dev dignified.js
+$ npm install --save-dev aegir.js
```
Comment (noffle) [inline js-project-guidelines.md]: I think it's just `aegir` on npm

File: js-project-guidelines.md [hunk 6]
```diff
@@ -191,53 +226,62 @@ If you install the module through npm and require it, you receive the ES5 versio
...
-We want to see the web move forward, and some of us enjoy writing their JavaScript with things like const and arrow functions.
+We want to see the web move forward, and some of us enjoy writing their JavaScript with things like `const` and arrow functions.
+
+#### Do I have to use ES2015 and Babel and aegir.js in my project?
+
+No.
+
+#### Do I have to bundle everything with webpack?
+
+No. But other people might ask you to at some point, so, maybe it's better to be prepared?
```
Comment (noffle) [inline js-project-guidelines.md]: Nit: change question (uncertainty) to a statement ("it's better to be prepared!").

File: js-project-guidelines.md [hunk 7]
```diff
@@ -191,53 +226,62 @@ If you install the module through npm and require it, you receive the ES5 versio
...
 
-#### Why are doing this?
+#### Why are you doing this?
```
Comment (noffle) [inline js-project-guidelines.md]: "This tooling was borne of hard learning done from years in the trenches. Its goal is to minimize process road bumps and provide a unified low-friction workflow for contributors."

File: js-project-guidelines.md [hunk 8]
```diff
@@ -191,53 +226,62 @@ If you install the module through npm and require it, you receive the ES5 versio
...
 Because it saves us hours every single day. Hours in which we don’t have to think about these things or argue with someone about why we are doing it.
 
-# Contributing
+## Contributing
 
-Follow these conventions described in this document.
+Please follow the conventions described in this document.
 
-When reporting a bug, if possible, provide a way to reproduce or even better, write a test that fails with your case
+When reporting a bug, if possible, provide a way to reproduce (Or even better, write a test that fails with your case).
```
Comment (noffle) [inline js-project-guidelines.md]: Nit: lower case 'o' within parens, right?

File: js-project-guidelines.md [hunk 9]
```diff
@@ -191,53 +226,62 @@ If you install the module through npm and require it, you receive the ES5 versio
...
 
-# References - Resources and good reads
+## References - Resources and good reads
 
 - Comparission between WebPack, browserify, requirejs, jspm and rollup - [https://github.com/webpack/docs/wiki/comparison](https://github.com/webpack/docs/wiki/comparison)
 - [The cost of transpiling ES2015 in 2016](https://github.com/samccone/The-cost-of-transpiling-es2015-in-2016)
 - [standardjs.com](http://standardjs.com/)
 
-# Acknowledgment
+## Acknowledgment
```
Comment (noffle) [inline js-project-guidelines.md]: Nit: missing trailing period

**Comment by noffle** [comment]
This is all great stuff, which is why my comments are all quibbles and nitpicks.

**Comment by diasdavid** [comment]
This looks great to me :) Other then the small nits and `s/aegir.js/aergir/g`, is RTM :) 

Thank you for reviewing this throughly, @RichardLitt :)

**Pull request merged**